examples/spisd/sd.c

```diff
@@ -85,8 +85,7 @@ static int sd_parse_csd(sd_card_info_t *ci, const uint32_t *bits)
 	csd->dsr_implemented = (bits[1] >> 12) & 0x1;
 
 	if (ci->type == sdCardType_SD1_x || ci->type == sdCardType_SD2_0) {
-		csd->device_size = (bits[1] << 2) & 0xffc;
-		csd->device_size |= (bits[2] >> 30) & 0x3;
```

Comment:
In what sense does these changes "fix size calculation for 32GB+ SD cards"?

If I'm not reading it wrong there is no effective change?

Reply:
That is correct, it will effectively be the same. Just keeping the calculation style consistent with the one for SDHC cards below that is fixed - one pass calculation and not unnecessary large constants for the AND mask.

Reply:
Ah, now I see, the lines further down that were changed actually were fixing a bug, I missed that on first read. 👍